Algorithm.CSharp/ShortableProviderOrdersRejectedRegressionAlgorithm.cs

```diff
@@ -0,0 +1,225 @@
+                    throw new Exception($"Expected 1 successful order, found: {_ordersAllowed.Count}");
+                }
+                if (_ordersDenied.Count != 2)
+                {
+                    throw new Exception($"Expected 2 failed orders, found: {_ordersDenied.Count}");
+                }
+
+                var allowedOrder = _ordersAllowed[0];
+                var orderUpdate = new UpdateOrderFields()
+                {
+                    LimitPrice = 0.01m,
+                    Quantity = -1001,
+                    Tag = "Testing updating and exceeding maximum quantity"
+                };
+
+                var response = allowedOrder.Update(orderUpdate);
+                if (response.ErrorCode != OrderResponseErrorCode.ExceedsShortableQuantity)
+                {
+                    throw new Exception($"Expected order to fail due to exceeded shortable quantity, found: {response.ErrorCode.ToString()}");
+                }
+
+                var cancelResponse = allowedOrder.Cancel();
+                if (cancelResponse.IsError)
+                {
+                    throw new Exception("Expected to be able to cancel open order after bad qty update");
+                }
+
+                _invalidatedAllowedOrder = true;
+                _ordersDenied.Clear();
+                _ordersAllowed.Clear();
+                return;
+            }
+
+            if (!_invalidatedNewOrderWithPortfolioHoldings)
+            {
+                HandleOrder(MarketOrder(_spy, -1000)); // Should succeed, no holdings and no open orders to stop this
+                var spyShares = Portfolio[_spy].Quantity;
+                if (spyShares != -1000m)
+                {
+                    throw new Exception($"Expected -1000 shares in portfolio, found: {spyShares}");
+                }
+
+                HandleOrder(LimitOrder(_spy, -1, 0.01m)); // Should fail, portfolio holdings are at the max shortable quantity.
+                if (_ordersDenied.Count != 1)
+                {
+                    throw new Exception($"Expected limit order to fail due to existing holdings, but found {_ordersDenied.Count} failures");
+                }
+
+                _ordersAllowed.Clear();
+                _ordersDenied.Clear();
+
+                HandleOrder(MarketOrder(_aig, -1001));
+                if (_ordersAllowed.Count != 1)
+                {
+                    throw new Exception($"Expected market order of -1001 BAC to not fail");
+                }
+
+                _invalidatedNewOrderWithPortfolioHoldings = true;
+            }
+        }
+
+        private void HandleOrder(OrderTicket orderTicket)
+        {
+            if (orderTicket.SubmitRequest.Status == OrderRequestStatus.Error)
+            {
+                _ordersDenied.Add(orderTicket);
+                return;
+            }
+
+            _ordersAllowed.Add(orderTicket);
+        }
+
+        public class RegressionTestShortableProvider : IShortableProvider
```

Comment:
Some for this Shortable provider, let's use the real file-based mockup data at `shortable/testbrokerage`